script/modelling/space_shuttle_reentry_trajectory.jl

```diff
@@ -0,0 +1,281 @@
+#' ---
+#' title: Space Shuttle Reentry Trajectory
+#' ---
+
+#' **Originally Contributed by**: Henrique Ferrolho
+
+#' This tutorial demonstrates how to compute a reentry trajectory for the
+#' [Space Shuttle](https://en.wikipedia.org/wiki/Space_Shuttle),
+#' by formulating and solving a nonlinear programming problem.
+#' The problem was drawn from Chapter 6 of 
+#' ["Practical Methods for Optimal Control and Estimation Using Nonlinear Programming"](https://epubs.siam.org/doi/book/10.1137/1.9780898718577),
+#' by John T. Betts.
+
+#' The motion of the vehicle is defined by the following set of DAEs:
+#' 
+#' $$
+#' \begin{align}
+#' \dot{h} & = v \sin \gamma , \\
+#' \dot{\phi} & = \frac{v}{r} \cos \gamma \sin \psi / \cos \theta , \\
+#' \dot{\theta} & = \frac{v}{r} \cos \gamma \cos \psi , \\
+#' \dot{v} & = -\frac{D}{m} - g \sin \gamma , \\
+#' \dot{\gamma} & = \frac{L}{m v} \cos(\beta) + \cos \gamma \left ( \frac{v}{r} - \frac{g}{v} \right ) , \\
+#' \dot{\psi} & = \frac{1}{m v \cos \gamma} L \sin(\beta) + \frac{v}{r \cos \theta} \cos \gamma \sin \psi \sin \theta , \\
+#' q & \le q_U , \\
+#' \end{align}
+#' $$
+
+#' where the aerodynamic heating on the vehicle wing leading edge is $q = q_a q_r$
+#' and the dynamic variables are
+#' 
+#' $$
+#' \begin{align}
+#'      h & \quad \text{altitude (ft)},     \qquad & & \gamma \quad \text{flight path angle (rad)}, \\
+#'   \phi & \quad \text{longitude (rad)},   \qquad & & \psi   \quad \text{azimuth (rad)},           \\
+#' \theta & \quad \text{latitude (rad)},    \qquad & & \alpha \quad \text{angle of attack (rad)},   \\
+#'      v & \quad \text{velocity (ft/sec)}, \qquad & & \beta  \quad \text{bank angle (rad)}.
+#' \end{align}
+#' $$
+
+#' The aerodynamic and atmospheric forces on the vehicle are
+#' specified by the following quantities (English units):
```

Comment:
I was not aware that this example is using English units :-p

Reply:
It is. :slightly_smiling_face: I implemented the problem as close as possible to the book version.

Reply:
Actually, I copied almost word-by-word the problem description from the book to the start of the notebook. (I.e., the markdown content of the current version.)

![image](https://user-images.githubusercontent.com/4166093/107356308-89a18700-6ac8-11eb-8f49-3da2fc74e6ed.png)

I hope this is not an issue. I do cite the book immediately at the start, so proper credit is given.

Reply:
I don't think this is an issue, as you stated, the book is correctly cited :-)